hazelcast/src/main/java/com/hazelcast/jet/impl/operation/InitExecutionOperation.java

```diff
@@ -140,16 +141,16 @@ private ExecutionPlan deserializePlan(Data planBlob) {
             return getNodeEngine().getSerializationService().toObject(planBlob);
         } else {
             JetServiceBackend service = getJetServiceBackend();
-            JobExecutionService jobExecutionService = service.getJobExecutionService();
+            JobConfig jobConfig = service.getJobConfig(jobId());
+            JobClassLoaderService jobClassloaderService = service.getJobClassLoaderService();
 
             try {
-                ClassLoader cl = service.getClassLoader(jobId());
-                JobConfig jobConfig = service.getJobConfig(this.jobId());
+                ClassLoader cl = jobClassloaderService.getOrCreateClassLoader(jobConfig, jobId(), EXECUTION);
```

Comment:
Not a big difference, but this line should come before the `try` statement. Then it better shows what the code in `finally` cleans up.